spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -239,6 +260,145 @@ public void setAutoStartup(boolean autoStartup) {
 		this.autoStartup = autoStartup;
 	}
 
+	public ErrorHandler getErrorHandler() {
+		return this.errorHandler;
+	}
+
+	public void setErrorHandler(ErrorHandler errorHandler) {
+		this.errorHandler = errorHandler;
+	}
+
+	protected AsyncTaskExecutor getConsumerTaskExecutor() {
+		return this.consumerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that poll the consumer. If the instance set is not a
+	 * {@link AsyncTaskExecutor} it will be wrapped into one.
+	 * @param consumerTaskExecutor the executor
+	 */
+	public void setConsumerTaskExecutor(Executor consumerTaskExecutor) {
+		if (consumerTaskExecutor instanceof AsyncTaskExecutor) {
+			this.consumerTaskExecutor = (AsyncTaskExecutor) consumerTaskExecutor;
+		}
+		else {
+			this.consumerTaskExecutor = new ConcurrentTaskExecutor(consumerTaskExecutor);
+		}
+	}
+
+	protected AsyncTaskExecutor getListenerTaskExecutor() {
+		return this.listenerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that invoke the listener. If the instance set is not a
+	 * {@link AsyncTaskExecutor} it will be wrapped into one.
+	 * @param listenerTaskExecutor the executor.
+	 */
+	public void setListenerTaskExecutor(Executor listenerTaskExecutor) {
+		if (listenerTaskExecutor instanceof AsyncTaskExecutor) {
+			this.listenerTaskExecutor = (AsyncTaskExecutor) listenerTaskExecutor;
+		}
+		else {
+			this.listenerTaskExecutor = new ConcurrentTaskExecutor(listenerTaskExecutor);
```

Comment:
I don't see reason to do all this adaptation. If we really need `AsyncTaskExecutor` let's just request it from the configuration level!
All Spring target `Executor` implementations are `AsyncTaskExecutor`.
Otherwise it looks weird. 
You request from me `Executor`, but when I request `getListenerTaskExecutor()` I receive `AsyncTaskExecutor`.
If your bizarre `Executor` (e.g. from Reactor) isn't like `AsyncTaskExecutor` you always can wrap it into `ConcurrentTaskExecutor` on your own.
But I don't see reason to do that in our code when we can be more strict.